src/mender-update/daemon/states.cpp

```diff
@@ -174,45 +193,21 @@ void SubmitInventoryState::DoSubmitInventory(Context &ctx, sm::EventPoster<State
 	}
 }
 
-void SubmitInventoryState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
-	// Schedule timer for next update first, so that long running submissions do not postpone
-	// the schedule.
-	log::Debug(
-		"Scheduling the next inventory submission in: "
-		+ to_string(ctx.mender_context.GetConfig().inventory_poll_interval_seconds) + " seconds");
-	retry_.wait_timer.AsyncWait(
-		chrono::seconds(ctx.mender_context.GetConfig().inventory_poll_interval_seconds),
-		[&poster](error::Error err) {
-			if (err != error::NoError) {
-				if (err.code != make_error_condition(errc::operation_canceled)) {
-					log::Error("Inventory poll timer caused error: " + err.String());
-				}
-			} else {
-				poster.PostEvent(StateEvent::InventoryPollingTriggered);
-			}
-		});
-
-	DoSubmitInventory(ctx, poster);
```

Comment:
Where is this happening now?

Reply:
in `OnEnter`. The previous `OnEnter` only scheduled the next interval and called `DoSubmitInventory`

Reply:
I see. I was looking for this function call somewhere. :+1: